packages/core/src/storage/StorageService.ts

```diff
@@ -10,7 +11,7 @@ interface AdvancedQuery<T extends BaseRecord> {
   $not?: Query<T>
 }
 
-export type Query<T extends BaseRecord> = AdvancedQuery<T> | SimpleQuery<T>
+export type Query<T extends BaseRecord<any, any, any>> = AdvancedQuery<T> | SimpleQuery<T>
```

Comment:
This is causing the validation error in the CI. What's the reason for this change over the previous Query<T extends BaseRecord>

Reply:
I was getting type error when trying to set the type of the query parameter to Query<ConnectionRecord>, Query<CredentialRecord> etc